File: shared/src/commonTest/kotlin/com/megabreezy/breezybuilds_wordle/core/ui/app_modal/mock/AppModalCommonMock.kt
package com.megabreezy.breezybuilds_wordle.core.ui.app_modal.mock

import com.megabreezy.breezybuilds_wordle.core.ui.app_modal.AppModalContentRepresentable
import com.megabreezy.breezybuilds_wordle.core.ui.app_modal.AppModalRepresentable
import com.megabreezy.breezybuilds_wordle.core.ui.app_modal.AppModalViewHandleable

class AppModalCommonMock: AppModalRepresentable
{
    var contentToReturn: AppModalContentRepresentable? = null

    var viewHandler: AppModalViewHandleable? = null

    override fun content(): AppModalContentRepresentable? = this.contentToReturn

    override fun handler(): AppModalViewHandleable? = viewHandler

    override fun setContent(newContent: AppModalContentRepresentable?) { this.contentToReturn = newContent }

    override fun setHandler(newHandler: AppModalViewHandleable?) { this.viewHandler = newHandler }
}

File: iosApp/iosAppTests/core/ui/app_modal/mock/AppModalCommonMock.swift
//
//  AppModalCommonMock.swift
//  iosAppTests
//
//  Created by Bradley Phillips on 4/9/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import shared

class AppModalCommonMock: AppModalRepresentable
{
    var contentToReturn: AppModalContentRepresentable? = nil
    var viewHandler: AppModalViewHandleable? = nil
    var setContentMethodPassedInNewContent: AppModalContentRepresentable? = nil
    var setHandlerMethodPassedInNewHandler: AppModalViewHandleable?  = nil
    
    func content() -> AppModalContentRepresentable? { contentToReturn }
    
    func handler() -> AppModalViewHandleable? { viewHandler }
    
    func setContent(newContent: AppModalContentRepresentable?)
    {
        self.setContentMethodPassedInNewContent = newContent
    }
    
    func setHandler(newHandler: AppModalViewHandleable?)
    {
        viewHandler = newHandler
        self.setHandlerMethodPassedInNewHandler = newHandler
    }
}
